architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,296 @@
+
+By default, the communication with registries use TLS. We allow plain text download of the recipes with locally hosted registry which might not have TLS capabilities. This is not recommended unless absolutely neccessary, to avoid any potential modification to the recipes.  
+
+#### Terraform Recipe execution
+
+Terraform recipes are download from internet too. We support public and private modules as well as different kinds of authentication. Terraform modules can be downloaded only on TLS.
+
+Terraform downloads providers from the internet onto an empty directory `/terraform` which is mounted into the applications RP pod and executes them. This means that these providers have access to the network, filesystem, and environment variables, as well as the running memory of Applications RP. Terraform communicates with AWS as needed to deploy resources.
+
+Terraform containerization efforts are in progress, which aims to overcome the above mentioned security challenges.
+
+#### Data Serialization / Formats
+
+We use custom parsers to parse Radius-related resource IDs and do not use any other custom parsers. Radius resource ids are a custom string format that can be parsed from untrusted data. The parser is a shared component.
+
+We also use json for data interchange between Applications RP and other services. We use golang standard libraries to parse json. 
+
+## Clients
+
+**UCP** UCP is the client of Applications RP.  It forwards all requests related to Applications and several of Application resources deployment to Applications RP.
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the Controller component:
+
+- **Kubernetes Cluster**: The overall environment where the Applications RP  operates and receives requests from the UCP.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads.
+
+The Applications RP component lives inside the `radius-system` namespace in the Kubernetes cluster where it is installed. UCP also resides within the same namespace.
+
+The Kubernetes API Server, with which Applications RP interacts, runs in the `kube-system` namespace within the cluster.
+
+Today, the RP assumes that incoming requests are from a trusted user who is authorized to make the request. It should enforce this by checking that the requests are made by UCP. UCP in turn should accept requests from authorized and authenticated users and clients such as `rad cli`. Once we have Radius RBAC support, we should configure UCP so that it accepts requests only from authorized and authenticated users.
```

Comment:
Can a client directly call Applications RP?

Reply:
if it impersonates UCP, then it can.